app/models/vm/operations.rb

```diff
@@ -14,7 +14,15 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    return if ipaddresses.blank?
+    if MiqRegion.my_region.role_active?("cockpit_ws")
+      uri = URI(MiqCockpit::WS::MIQ_REDIRECT_URL)
```

Comment:
Why aren't we setting the `host` for this URI to be the hostname/ipaddress of the appliance with the cockpit role active?  Because we're not setting the `host` here to the cockpit appliance, all appliances need to have the local proxy in apache know which appliance is the one with cockpit.  I believe this is why we are installing the apache configs for all appliances via the `install_apache_proxy_config` method.

Is this because we have a valid session/token on the current appliance and can't use that session on the cockpit appliance?